library/cipher.c

```diff
@@ -1195,9 +1201,7 @@ int mbedtls_cipher_check_tag( mbedtls_cipher_context_t *ctx,
 
         /* Check the tag in "constant-time" */
         if( mbedtls_ct_memcmp( tag, check_tag, tag_len ) != 0 )
-            return( MBEDTLS_ERR_CIPHER_AUTH_FAILED );
-
-        return( 0 );
+            ret = MBEDTLS_ERR_CIPHER_AUTH_FAILED;
```

Comment:
Nit: This assumes there are no other statements on this code path. Could use an exit label to make it robust to refactoring? (Also line 1224.)